src/Service/AccessManagement/ResolveAllUrls.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Lullabot\Mpx\Service\AccessManagement;
+
+use Lullabot\Mpx\Service\IdentityManagement\UserSession;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Resolve all URLs for a given service.
+ *
+ * @see https://docs.theplatform.com/help/wsf-resolveallurls-method
+ */
+class ResolveAllUrls
+{
+    /**
+     * While this method is not a data service, it still has a schema version.
+     */
+    const SCHEMA_VERSION = '1.0';
+
+    /**
+     * The URL of the method. Note that we hardcode the whole path as this is
+     * where other services are bootstrapped from.
+     */
+    const RESOLVE_ALL_URLS_URL = 'https://access.auth.theplatform.com/web/Registry/resolveAllUrls';
+
+    /**
+     * An array of resolved URLs for the service.
+     *
+     * @var string[]
+     */
+    protected $resolved;
+
+    /**
+     * The service these URLs correspond to, such as 'Media Data Service'.
+     *
+     * @var string
+     */
+    protected $service;
+
+    /**
+     * ResolveAllUrls constructor.
+     *
+     * @todo Is there value in storing Responses in all classes?
+     *
+     * @param string $service The service that was queried, such as 'Access Data Service'.
+     * @param array  $data    The MPX response.
+     */
+    public function __construct(string $service, array $data)
+    {
+        if (!isset($data['resolveAllUrlsResponse'])) {
+            throw new \InvalidArgumentException('Data does not contain a resolveAllUrlsResponse key and does not appear to be an MPX response.');
+        }
+
+        $this->resolved = $data['resolveAllUrlsResponse'];
```

Comment:
true? It's an array of resolved URLs such as:

```
{
    "resolveAllUrlsResponse": [
        "http://mps.theplatform.com",
        "http://data.media.theplatform.com/media"
    ]
}
```

Reply:
Sorry, I thought you were storing that just for the sake of a boolean check, meaning: is this resolved? Not actually because we might need to retrieve the result stored there. The word confused me.

Reply:
Ah, got it! I will rename the method and variable to `getResolvedUrls` tomorrow.

Reply:
oh! that's perfect! thanks